flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBResourceContainer.java

```diff
@@ -143,6 +144,11 @@ public ColumnFamilyOptions getColumnOptions() {
 					"We currently only support BlockBasedTableConfig When bounding total memory.");
 				blockBasedTableConfig = (BlockBasedTableConfig) tableFormatConfig;
 			}
+			if (rocksResources.isUsingPartitionedIndex()) {
```

Comment:
From RocksDB's current implementation, partitioned index is only compatible with full bloom filter. From my point of view, we should check whether current filter is full instead of block based and set a default full bloom filter if current filter is not `null`.

Reply:
We couldn't know the filter config(bits, useBlockBasedMode) if it's already created. Should we overwrite with a default full bloom filter? It might have not expected behavior.

Reply:
I think we could overwrite a default full bloom filter if user configured a filter as this is required for partitioned index feature. We could add warning logs for this behavior change. Remember to add a `TODO` so that we could get the `bits` in configured bloom filter in the future once RocksDB bumped version.